src/uu/mktemp/src/mktemp.rs

```diff
@@ -137,64 +137,53 @@ struct Options {
-///
-/// This function is required to work around a limitation of `clap`,
-/// the command-line argument parsing library. In case the command
-/// line is
-///
-/// ```sh
-/// mktemp --tmpdir XXX
-/// ```
-///
-/// the program should behave like
-///
-/// ```sh
-/// mktemp --tmpdir=${TMPDIR:-/tmp} XXX
-/// ```
-///
-/// However, `clap` thinks that `XXX` is the value of the `--tmpdir`
-/// option. This function returns `true` in this case and `false`
-/// in all other cases.
-fn is_tmpdir_argument_actually_the_template(matches: &ArgMatches) -> bool {
-    if !matches.contains_id(ARG_TEMPLATE) {
-        if let Some(tmpdir) = matches.get_one::<String>(OPT_TMPDIR) {
-            if !Path::new(tmpdir).is_dir() && tmpdir.contains("XXX") {
-                return true;
-            }
-        }
-    }
-    false
-}
-
 impl Options {
     fn from(matches: &ArgMatches) -> Self {
-        // Special case to work around a limitation of `clap`; see
-        // `is_tmpdir_argument_actually_the_template()` for more
-        // information.
-        //
-        // Fixed in clap 3
-        // See https://github.com/clap-rs/clap/pull/1587
-        let (tmpdir, template) = if is_tmpdir_argument_actually_the_template(matches) {
-            let tmpdir = Some(env::temp_dir().display().to_string());
-            let template = matches.get_one::<String>(OPT_TMPDIR).unwrap().to_string();
-            (tmpdir, template)
-        } else {
-            // If no template argument is given, `--tmpdir` is implied.
-            match matches.get_one::<String>(ARG_TEMPLATE) {
-                None => {
-                    let tmpdir = match matches.get_one::<String>(OPT_TMPDIR) {
-                        None => Some(env::temp_dir().display().to_string()),
-                        Some(tmpdir) => Some(tmpdir.to_string()),
-                    };
-                    let template = DEFAULT_TEMPLATE;
-                    (tmpdir, template.to_string())
-                }
-                Some(template) => {
-                    let tmpdir = matches.get_one::<String>(OPT_TMPDIR).map(String::from);
-                    (tmpdir, template.to_string())
-                }
+        // Workaround due to way clap handles arguments.
+        // Short and long tmpdir options have different behavior based on how arguments are given
+        // Due to limitations in clap, this behavior is impossible using a single flag
+        // Thus, a short option (-p) is additionally created in addition
+        // to the long option(--tmpdir)
+        let short_tmpdir = match matches.get_one::<String>(OPT_SHORT_TMPDIR) {
+            Some(tmpdir) => Some(tmpdir.to_string()),
+            None => Some(env::temp_dir().display().to_string()),
+        };
+
+        let long_tmpdir = match matches.get_one::<String>(OPT_TMPDIR) {
+            Some(tmpdir) => Some(tmpdir.to_string()),
+            None => Some(env::temp_dir().display().to_string()),
```

Comment:
If neither `-p` nor `--tempdir` is supplied, then we needlessly call `env::temp_dir().display().to_string()` twice.